c7n/resources/s3.py

```diff
@@ -117,8 +117,8 @@ def augment(self, buckets):
     ('get_bucket_versioning', 'Versioning', None, None),
     ('get_bucket_website', 'Website', None, None),
     ('get_bucket_logging', 'Logging', None, 'LoggingEnabled'),
-    ('get_bucket_notification_configuration', 'Notification', None, None)
-#        ('get_bucket_lifecycle', 'Lifecycle', None, None),
+    ('get_bucket_notification_configuration', 'Notification', None, None),
+    ('get_bucket_lifecycle', 'Lifecycle', None, None)
```

Comment:
just to note effectively by this one line uncommented, you can then do

```
  - type: value
     key: Notification[].AbortIncompleteMultipartUpload
     value: not-null
```

to get the same effect as the custom filter here.